tests/frame_test.go

```diff
@@ -72,3 +74,46 @@ func TestFrameDismissDialogBox(t *testing.T) {
 		})
 	}
 }
+
+func TestFrameNoPanicWithEmbeddedIFrame(t *testing.T) {
```

Comment:
This test races for me on Linux:

<details>
  <summary>Race</summary>

```
$ XK6_HEADLESS= go test -v -race -failfast -count=1 -run 'TestFrameNoPanicWithEmbeddedIFrame' ./tests
=== RUN   TestFrameNoPanicWithEmbeddedIFrame
=== PAUSE TestFrameNoPanicWithEmbeddedIFrame
=== CONT  TestFrameNoPanicWithEmbeddedIFrame
time="2022-12-20T17:21:06+01:00" level=warning msg="url:https://www.youtube.com/embed/gwO7k5RTE54?wmode=opaque&enablejsapi=1 method:GET err:fetching response body: No resource with given identifier found (-32000)" category="Response:bodySize:fetchBody" elapsed="0 ms" goroutine=152
time="2022-12-20T17:21:06+01:00" level=warning msg="url:https://www.youtube.com/s/player/34f9b71c/www-player.css method:GET err:fetching response body: No resource with given identifier found (-32000)" category="Response:bodySize:fetchBody" elapsed="0 ms" goroutine=152
time="2022-12-20T17:21:06+01:00" level=warning msg="url:https://www.youtube.com/s/player/34f9b71c/fetch-polyfill.vflset/fetch-polyfill.js method:GET err:fetching response body: No resource with given identifier found (-32000)" category="Response:bodySize:fetchBody" elapsed="0 ms" goroutine=152
time="2022-12-20T17:21:06+01:00" level=warning msg="url:https://www.youtube.com/s/player/34f9b71c/www-embed-player.vflset/www-embed-player.js method:GET err:fetching response body: No resource with given identifier found (-32000)" category="Response:bodySize:fetchBody" elapsed="0 ms" goroutine=152
time="2022-12-20T17:21:06+01:00" level=warning msg="url:https://fonts.gstatic.com/s/roboto/v18/KFOmCnqEu92Fr1Mu4mxK.woff2 method:GET err:fetching response body: No resource with given identifier found (-32000)" category="Response:bodySize:fetchBody" elapsed="0 ms" goroutine=152
time="2022-12-20T17:21:06+01:00" level=warning msg="url:https://fonts.gstatic.com/s/roboto/v18/KFOlCnqEu92Fr1MmEU9fBBc4.woff2 method:GET err:fetching response body: No resource with given identifier found (-32000)" category="Response:bodySize:fetchBody" elapsed="0 ms" goroutine=152
time="2022-12-20T17:21:06+01:00" level=warning msg="url:https://www.youtube.com/s/player/34f9b71c/player_ias.vflset/en_US/base.js method:GET err:fetching response body: No resource with given identifier found (-32000)" category="Response:bodySize:fetchBody" elapsed="0 ms" goroutine=152
time="2022-12-20T17:21:06+01:00" level=error msg="Failed to load resource: net::ERR_SSL_KEY_USAGE_INCOMPATIBLE" browser_source=network line_number=0 source=browser stacktrace="<nil>" url="https://googleads.g.doubleclick.net/pagead/id"
time="2022-12-20T17:21:06+01:00" level=error msg="Failed to load resource: net::ERR_SSL_KEY_USAGE_INCOMPATIBLE" browser_source=network line_number=0 source=browser stacktrace="<nil>" url="https://static.doubleclick.net/instream/ad_status.js"
time="2022-12-20T17:21:06+01:00" level=warning msg="url:https://www.google.com/js/th/YkfklCtf3s5-_1quWHAnTHHVaBZ-i7ToAeXFpu3i2Ro.js method:GET err:fetching response body: No resource with given identifier found (-32000)" category="Response:bodySize:fetchBody" elapsed="0 ms" goroutine=152
time="2022-12-20T17:21:06+01:00" level=warning msg="url:https://www.youtube.com/s/player/34f9b71c/player_ias.vflset/en_US/embed.js method:GET err:fetching response body: No resource with given identifier found (-32000)" category="Response:bodySize:fetchBody" elapsed="0 ms" goroutine=152
time="2022-12-20T17:21:06+01:00" level=warning msg="url:https://www.youtube.com/s/player/34f9b71c/player_ias.vflset/en_US/remote.js method:GET err:fetching response body: No resource with given identifier found (-32000)" category="Response:bodySize:fetchBody" elapsed="0 ms" goroutine=152
time="2022-12-20T17:21:07+01:00" level=warning msg="url:https://yt3.ggpht.com/9mD5UVq9v0F0JsJYQN-zMB8AfV2Wi4SOdF7B2AAxlfekZWgitfj9ewjXhvWmGYJe3RIff0gXWw=s68-c-k-c0x00ffffff-no-rj method:GET err:fetching response body: No resource with given identifier found (-32000)" category="Response:bodySize:fetchBody" elapsed="0 ms" goroutine=152
time="2022-12-20T17:21:07+01:00" level=warning msg="url:https://i.ytimg.com/vi_webp/gwO7k5RTE54/default.webp method:GET err:fetching response body: No resource with given identifier found (-32000)" category="Response:bodySize:fetchBody" elapsed="0 ms" goroutine=152
time="2022-12-20T17:21:07+01:00" level=warning msg="url:https://jnn-pa.googleapis.com/$rpc/google.internal.waa.v1.Waa/Create method:POST err:fetching response body: No resource with given identifier found (-32000)" category="Response:bodySize:fetchBody" elapsed="0 ms" goroutine=152
time="2022-12-20T17:21:07+01:00" level=warning msg="url:https://www.youtube.com/generate_204?_Z3IgA method:GET err:fetching response body: No resource with given identifier found (-32000)" category="Response:bodySize:fetchBody" elapsed="0 ms" goroutine=152
time="2022-12-20T17:21:07+01:00" level=warning msg="url:https://jnn-pa.googleapis.com/$rpc/google.internal.waa.v1.Waa/GenerateIT method:POST err:fetching response body: No resource with given identifier found (-32000)" category="Response:bodySize:fetchBody" elapsed="0 ms" goroutine=152
time="2022-12-20T17:21:07+01:00" level=warning msg="url:https://www.gstatic.com/cv/js/sender/v1/cast_sender.js method:GET err:fetching response body: No resource with given identifier found (-32000)" category="Response:bodySize:fetchBody" elapsed="0 ms" goroutine=152
time="2022-12-20T17:21:07+01:00" level=warning msg="url:https://www.gstatic.com/eureka/clank/107/cast_sender.js method:GET err:fetching response body: No resource with given identifier found (-32000)" category="Response:bodySize:fetchBody" elapsed="0 ms" goroutine=152
==================
WARNING: DATA RACE
Write at 0x00c00050e570 by goroutine 50:
  runtime.mapassign_faststr()
      /home/ivan/.local/share/go/src/runtime/map_faststr.go:203 +0x0
  github.com/grafana/xk6-browser/common.(*FrameManager).requestStarted()
      /home/ivan/Projects/grafana/xk6-browser/common/frame_manager.go:522 +0x2e9
  github.com/grafana/xk6-browser/common.(*NetworkManager).onRequest()
      /home/ivan/Projects/grafana/xk6-browser/common/network_manager.go:441 +0xa1d
  github.com/grafana/xk6-browser/common.(*NetworkManager).handleEvents()
      /home/ivan/Projects/grafana/xk6-browser/common/network_manager.go:334 +0x2e9
  github.com/grafana/xk6-browser/common.(*NetworkManager).initEvents.func1()
      /home/ivan/Projects/grafana/xk6-browser/common/network_manager.go:313 +0x3e

Previous write at 0x00c00050e570 by goroutine 171:
  runtime.mapdelete_faststr()
      /home/ivan/.local/share/go/src/runtime/map_faststr.go:301 +0x0
  github.com/grafana/xk6-browser/common.(*FrameManager).requestFinished()
      /home/ivan/Projects/grafana/xk6-browser/common/frame_manager.go:490 +0x1e9
  github.com/grafana/xk6-browser/common.(*NetworkManager).onLoadingFinished()
      /home/ivan/Projects/grafana/xk6-browser/common/network_manager.go:389 +0x664
  github.com/grafana/xk6-browser/common.(*NetworkManager).handleEvents()
      /home/ivan/Projects/grafana/xk6-browser/common/network_manager.go:332 +0x1ce
  github.com/grafana/xk6-browser/common.(*NetworkManager).initEvents.func1()
      /home/ivan/Projects/grafana/xk6-browser/common/network_manager.go:313 +0x3e

Goroutine 50 (running) created at:
  github.com/grafana/xk6-browser/common.(*NetworkManager).initEvents()
      /home/ivan/Projects/grafana/xk6-browser/common/network_manager.go:312 +0x34e
  github.com/grafana/xk6-browser/common.NewNetworkManager()
      /home/ivan/Projects/grafana/xk6-browser/common/network_manager.go:87 +0x5e4
  github.com/grafana/xk6-browser/common.NewFrameSession()
      /home/ivan/Projects/grafana/xk6-browser/common/frame_session.go:100 +0x73a
  github.com/grafana/xk6-browser/common.NewPage()
      /home/ivan/Projects/grafana/xk6-browser/common/page.go:119 +0xc18
  github.com/grafana/xk6-browser/common.(*Browser).onAttachedToTarget()
      /home/ivan/Projects/grafana/xk6-browser/common/browser.go:274 +0xadb
  github.com/grafana/xk6-browser/common.(*Browser).initEvents.func1()
      /home/ivan/Projects/grafana/xk6-browser/common/browser.go:175 +0x424

Goroutine 171 (running) created at:
  github.com/grafana/xk6-browser/common.(*NetworkManager).initEvents()
      /home/ivan/Projects/grafana/xk6-browser/common/network_manager.go:312 +0x34e
  github.com/grafana/xk6-browser/common.NewNetworkManager()
      /home/ivan/Projects/grafana/xk6-browser/common/network_manager.go:87 +0x5e4
  github.com/grafana/xk6-browser/common.NewFrameSession()
      /home/ivan/Projects/grafana/xk6-browser/common/frame_session.go:100 +0x73a
  github.com/grafana/xk6-browser/common.(*FrameSession).attachIFrameToTarget()
      /home/ivan/Projects/grafana/xk6-browser/common/frame_session.go:847 +0x266
  github.com/grafana/xk6-browser/common.(*FrameSession).onAttachedToTarget()
      /home/ivan/Projects/grafana/xk6-browser/common/frame_session.go:780 +0x476
  github.com/grafana/xk6-browser/common.(*FrameSession).initEvents.func1()
      /home/ivan/Projects/grafana/xk6-browser/common/frame_session.go:259 +0x608
==================
time="2022-12-20T17:21:07+01:00" level=error msg="Failed to load resource: the server responded with a status of 404 (Not Found)" browser_source=network line_number=0 source=browser stacktrace="<nil>" url="http://127.0.0.1:39275/favicon.ico"
    testing.go:1319: race detected during execution of test
--- FAIL: TestFrameNoPanicWithEmbeddedIFrame (1.61s)
FAIL
FAIL    github.com/grafana/xk6-browser/tests    1.652s
```
</details>

It doesn't seem related to the changes here, and it also happens on `main`, with _many_ more data races, so at least it's an improvement? :smile:

Reply:
😩 we run into new data race issues a lot still. I've created issue [#680](https://github.com/grafana/xk6-browser/issues/680) to track it.